### DISCUSSION TITLE
Help! Code to make a background img in the body that is stored stored locally

### DISCUSSION THREAD

**Discussion by Margaret6556** [twbs/bootstrap#36709] [Q&A] [answered]
How to code a background img in the CSS body from a local file NOT attached to an external URL? I can't find the code for bootstrap V4.6 and later. 
This doesn't work....without >> https://

flex-container {
  background-image: url("Background1.jpg");
}

TIA.

<img width="961" alt="Screenshot html-css" src="https://user-images.githubusercontent.com/99307243/178087545-7aecf984-9bfb-436a-be9c-d2965804c45c.png">

**Comment by gerardnico** [comment]
With a local file uri ?

https://datacadamia.com/file/local#example_for_a_windows_path

It would not work on a http server.

**Reply by Margaret6556** [reply]
Thanks Gerard.  Yes, I wanted it saved in the local file system as png or jpg. Does bootstrap 5 allow that only on CSS or no?

**Reply by gerardnico** [reply]
Normally, you served your html file from a web server. 

The url may relative or absolute.

A local file uri is just a hack.

Boostrap has nothing to do with it.

I would recommend to use browsersync:
https://datacadamia.com/web/build/browser-sync

- places your html and image in the same directory, then starts brwosersync. Your first relative url should work.

**Reply by Margaret6556** [reply]
Thanks. I haven't found the soln yet.

**Comment by ale94lko** [comment]
@Margaret6556 

This is not related to ```Bootstrap```, it is pure ```CSS```.
I tested using [XAMPP](https://www.apachefriends.org/) as web server

# HTML CODE
```
<!DOCTYPE html>
<html>
    <head>
        <style>
            .flex-container {
                background-image: url("img/background.jpg");
            }
        </style>
    </head>
    <body>
        <div class="flex-container">
            <h1>This is a test</h1>
            <p>Hello World!</p>
        </div>
    </body>
</html>
```
  
# LOCAL URL
```http://localhost/test```

# DIRECTORY STRUCTURE
```
+- htdocs/
     |
     +- test/
         |
         +- img/
             |
             +- background.jpg
         |
         +- index.html
```

**Reply by Margaret6556** [reply]
Hi. Thanks. I'm not familiar with XAMPP.

**Comment by Margaret6556** [comment]
Hi Fidel,

Thanks, but it didn't work on bootstrap V4.6. 🤷‍♀️

Why do you think so?

[image: image.png]

Thanks

**Reply by ale94lko** [reply]
Let me try with V4.6.I'll Keep you posted.

**Comment by ale94lko** [comment]
@Margaret6556
Testing with V4.6 passed.

# HTML CODE
```
<!doctype html>
<html>
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

        <title>Hello, world!</title>

        <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" integrity="sha384-zCbKRCUGaJDkqS1kPbPd7TveP5iyJE0EjAuZQTgFLD2ylzuqKfdKlfG/eSrtxUkn" crossorigin="anonymous">

        <style>
            .flex-container {
                width: 90%;
                height: 100%;
                background-image: url("img/background.png");
            }
        </style>

        <script src="https://cdn.jsdelivr.net/npm/jquery@3.5.1/dist/jquery.slim.min.js" integrity="sha384-DfXdz2htPH0lsSSs5nCTpuj/zy4C+OGpamoFVy38MVBnE+IbbVYUew+OrCXaRkfj" crossorigin="anonymous"></script>
        <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.bundle.min.js" integrity="sha384-fQybjgWLrvvRgtW6bFlB7jaZrFsaBXjsOMm/tB9LTS58ONXgqbR9W8oWht/amnpF" crossorigin="anonymous"></script>
    </head>
    <body>
        <section class="jumbotron text-center">
            <div class="container flex-container">
                <h1>Example</h1>
                <p class="lead">Lorem ipsum dolor sit amet,
                    consectetur adipiscing elit, sed do eiusmod tempor
                    incididunt ut labore et dolore magna aliqua. Ut enim
                    ad minim veniam, quis nostrud exercitation ullamco
                    laboris nisi ut aliquip ex ea commodo consequat.
                    Duis aute irure dolor in reprehenderit in voluptate
                    velit esse cillum dolore eu fugiat nulla pariatur.
                </p>
                <p>
                    <a href="#" class="btn btn-primary my-2">Primary action</a>
                    <a href="#" class="btn btn-secondary my-2">Secondary action</a>
                </p>
            </div>
        </section>
    </body>
</html>
```

**Reply by ale94lko** [reply]
This is how it looks.

![image](https://user-images.githubusercontent.com/47539231/179147772-6cfde207-f6d6-4dab-aaac-a5f2e6d2d0bf.png)

**Comment by SamuelMiller** [comment, marked as answer]
As a side note, why not use Bootstrap v5? The stable 5.2 version is soon to be released.

**Reply by ale94lko** [reply]
Agreed, but sometimes a project is built in a version and migrate to an updated one takes some time.
